app/components/filter-plaques/component.js

```diff
@@ -1,21 +1,53 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({
+var ID_FILTERS = ['contributors', 'funders', 'publishers'];
 
+export default Ember.Component.extend({
+    filters: null,
     init () {
-        this._super(...arguments);
-        let parameters = this.get('parameters');
-        this.set('filters', Object.keys(parameters).map((key, idx, arr) => {
-            return {
-                "key": key,
-                "value": parameters[key]
-            }
-        }).filter((item) => {
-            if (item.key == "page") {
-                return false;
-            }
-            return true;
-        }));
+      this._super(...arguments);
+      let parameters = this.get('parameters');
+      var filters = Object.keys(parameters).filter((key) => {
+        return key !== "page";
+      }).map((key) => {
+        return {
+          "key": key,
+          "value": parameters[key]
+        }
+      });
+
+      // Find and format all query params that use ids for the value
+      var ids = filters.filter((item) => {
+          return ID_FILTERS.indexOf(item.key) !== -1;
+      }).map((param) => {
+          return {key: param.value}
+      });
+
+      // Fetch display names
+      if (ids.length > 0) {
+        this.fetchAgentDetails(ids).then((data) => {
+            var displayFilters = filters.map((filter) => {
+              var value = filter.value;
+              data.forEach((agentData) => {
+                if (value === agentData.id) {
+                  value = agentData.name;
```

Comment:
Relying on order seems fragile because as you add/remove filters, the order of the items in the filters array will change. Then, the index of the item in `agentDetail` will no longer match up to the index of the item in `filters`. 

E.g. If I filter the results by a contributor (e.g. contributors=64025-884-465) it does get replaced with "Scripps Institute of Oceanography": 

![screen shot 2017-05-10 at 10 46 27 am](https://cloud.githubusercontent.com/assets/6414394/25904954/bc1422e4-356e-11e7-979a-19426cf3843f.png)

But if I add another filter, such as `tags=field_expedition`, the order of the tags changes: 

![screen shot 2017-05-10 at 10 47 04 am](https://cloud.githubusercontent.com/assets/6414394/25904980/d0844a7e-356e-11e7-83b5-19ea98fc2ec6.png)